File: src/cviz/timeline/commands/PlayCommand.java
package cviz.timeline.commands;

import cviz.IProcessor;
import se.svt.caspar.amcp.AmcpLayer;

public class PlayCommand extends ICommand {
    public PlayCommand(int layerId) {
        super(layerId);
    }

    @Override
    public boolean execute(IProcessor processor) {
        AmcpLayer layer = processor.getLayer(getLayerId());

        try {
            layer.play();
            return true;
        } catch (Exception e){
            System.err.println("Failed to execute command: " + e.getMessage());
            return false;
        }
    }

    @Override
    public String toString() {
        return "PlayCommand: " + getLayerId();
    }
}

File: CViz/Timeline/Command/PlayCommand.cs
namespace CViz.Timeline.Command
{
    class PlayCommand : AmcpCommand
    {
        public PlayCommand(int layerId, string command) : base(layerId, command)
        {
        }

        public override bool Execute(ITimeline timeline)
        {
            if (!base.Execute(timeline))
                return false;

            LayerState state = timeline.GetLayerState(LayerId);
            if (state != null)
                state.LastFrame = 0;

            return true;
        }

        protected override string CommandName => "PlayCommand";
    }
}